src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,163 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private MyNode head;
+    private MyNode tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        MyNode myNode = new MyNode(value);
+        if (size == 0) {
+            head = myNode;
+            tail = myNode;
+        } else {
+            tail.setNext(myNode);
+            myNode.setPrev(tail);
+            tail = myNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Out of bounds");
+        }
+
+        MyNode myNode = new MyNode(value);
+
+        if (index == size) {
+
+            add(value);
+        } else if (index == 0) {
+            myNode.setNext(head);
+            head.setPrev(myNode);
+            head = myNode;
+            size++;
+        } else {
+
+            MyNode current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.getNext();
```

Comment:
Consider creating a `findNodeByIndex` method to avoid code duplication in node traversal as suggested in the checklist.